### PR TITLE
Glass jaw quirk

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

A quirk that makes any hit to your head have a chance to knock you out. Sharp hits are a bit worse at knocking out than blunt ones.
I always felt like this favorite CDDA trait mine was missing from the game, but I tried implementing it in a more unique way.

## Why It's Good For The Game

99 losses, 1 win. Winners don't give up.
glassjoe

## Changelog

:cl: Fat bugs bunny
add: Added a glass jaw quirk, that leaves you prone to being knocked out when hit on the head.
/:cl:

### CODE REVIEW

File: code/datums/quirks/negative_quirks.dm [hunk 1]
```diff
@@ -238,6 +238,41 @@
 	quirk_holder.clear_mood_event("family_heirloom_missing")
 	quirk_holder.clear_mood_event("family_heirloom")
 
+/datum/quirk/glassjaw
+	name = "Glass Jaw"
+	desc = "You have a very fragile jaw. Any sufficiently hard blow to your head might knock you out."
+	icon = "facegrimace"
+	value = -4
+	gain_text = "<span class='danger'>Your jaw feels loose.</span>"
+	lose_text = "<span class='notice'>Your jaw feels connected again.</span>"
```
Comment (san7890) [inline code/datums/quirks/negative_quirks.dm:247]:
```suggestion
	gain_text = span_danger("Your jaw feels loose.")
	lose_text = span_notice("Your jaw feels connected again.")
```

Comment (ChungusGamer666) [reply code/datums/quirks/negative_quirks.dm:247]: everything else uses manual spans but i'm sure that's not a good thing so sure.

**Comment by Oranggu** [comment]
Another classic chungusgamer PR 🥶

**Comment by moocowswag** [comment]
Now we are just missing a narcolepsy quirk and people can take these + heavy sleeper for the best ss13 experience.

**Comment by Overuseddart** [comment]
Yes.

File: code/datums/quirks/negative_quirks.dm [hunk 2]
```diff
@@ -238,6 +238,42 @@
...
+		return
+	var/actual_damage = damage - (damage * blocked/100)
+	//only roll for knockouts at 5 damage or more
+	if(actual_damage < 5)
+		return
+	//blunt items are more likely to knock out, but sharp ones are still capable of doing it
+	if(prob(CEILING(actual_damage * (sharpness & SHARP_EDGED|SHARP_POINTY ? 0.65 : 1), 1)))
+		source.visible_message(span_warning("[source] gets knocked out!"),
+							span_userdanger("You are knocked out!"),
+							vision_distance = COMBAT_MESSAGE_RANGE)
```
Comment (Kampfkarren) [inline code/datums/quirks/negative_quirks.dm:274]:
```suggestion
		source.visible_message(
			span_warning("[source] gets knocked out!"),
			span_userdanger("You are knocked out!"),
			vision_distance = COMBAT_MESSAGE_RANGE,
		)
```

Comment (Mothblocks) [inline code/datums/quirks/negative_quirks.dm:274]:
```suggestion
		source.visible_message(
			span_warning("[source] gets knocked out!"),
			span_userdanger("You are knocked out!"),
			vision_distance = COMBAT_MESSAGE_RANGE,
		)
```

File: code/modules/mob/living/carbon/human/species.dm [hunk 3]
```diff
@@ -1329,7 +1329,7 @@ GLOBAL_LIST_EMPTY(features_by_species)
 	return TRUE
 
 /datum/species/proc/apply_damage(damage, damagetype = BRUTE, def_zone = null, blocked, mob/living/carbon/human/H, forced = FALSE, spread_damage = FALSE, wound_bonus = 0, bare_wound_bonus = 0, sharpness = NONE, attack_direction = null)
-	SEND_SIGNAL(H, COMSIG_MOB_APPLY_DAMAGE, damage, damagetype, def_zone, wound_bonus, bare_wound_bonus, sharpness, attack_direction)
+	SEND_SIGNAL(H, COMSIG_MOB_APPLY_DAMAGE, damage, damagetype, def_zone, blocked, wound_bonus, bare_wound_bonus, sharpness, attack_direction)
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species.dm:1332]: Probably need to update the comment on this?

File: code/datums/quirks/negative_quirks.dm [hunk 4]
```diff
@@ -238,6 +238,42 @@
 	quirk_holder.clear_mood_event("family_heirloom_missing")
 	quirk_holder.clear_mood_event("family_heirloom")
 
+/datum/quirk/glassjaw
```
Comment (Mothblocks) [inline code/datums/quirks/negative_quirks.dm:241]: Nit: `glass_jaw`

File: code/datums/quirks/negative_quirks.dm [hunk 5]
```diff
@@ -238,6 +238,42 @@
...
+	name = "Glass Jaw"
+	desc = "You have a very fragile jaw. Any sufficiently hard blow to your head might knock you out."
+	icon = "glass"
```
Comment (Mothblocks) [inline code/datums/quirks/negative_quirks.dm:244]: Isn't this just a drinking glass? Gonna look weird, What about `bone-break`, do we use that?

File: code/modules/mob/living/carbon/human/species.dm (hunk 3 above)
Comment (Mothblocks) [reply code/modules/mob/living/carbon/human/species.dm:1332]: The comment on MOB_APPLY_DAMAGE that specifies its argumnets

Comment (Mothblocks) [reply code/modules/mob/living/carbon/human/species.dm:1332]: Also are you sure that this doesn't break any existing consumers

Comment (ChungusGamer666) [reply code/modules/mob/living/carbon/human/species.dm:1332]: I checked every single use case, none of them used arguments beyond def_zone

Comment (ChungusGamer666) [reply code/modules/mob/living/carbon/human/species.dm:1332]: I'll update the comment soon

**Comment by Wallemations** [comment]
In response to the last commit:

"Yea, one request; lose the shorts!"

File: code/datums/quirks/negative_quirks.dm [hunk 6]
```diff
@@ -238,6 +238,57 @@
...
+/datum/quirk/glass_jaw/proc/punch_out(mob/living/carbon/source, damage, damagetype, def_zone, blocked, wound_bonus, bare_wound_bonus, sharpness, attack_direction)
+	SIGNAL_HANDLER
+	if((damagetype != BRUTE) || (def_zone != BODY_ZONE_HEAD))
+		return
+	var/actual_damage = damage - (damage * blocked/100)
+	//only roll for knockouts at 5 damage or more
+	if(actual_damage < 5)
+		return
+	//blunt items are more likely to knock out, but sharp ones are still capable of doing it
+	if(prob(CEILING(actual_damage * (sharpness & SHARP_EDGED|SHARP_POINTY ? 0.65 : 1), 1)))
```
Comment (Mothblocks) [inline code/datums/quirks/negative_quirks.dm:284]:
```suggestion
	if(prob(CEILING(actual_damage * (sharpness & (SHARP_EDGED|SHARP_POINTY) ? 0.65 : 1), 1)))
```

Not sure how the ambiguity here is going to work, so let's just be safe

**Comment by ChungusGamer666** [comment]
> Your Fix/Feature pull request is currently below zero (-3). Maintainers may close future Feature/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

yeah go fuck yourself

**Pull request merged**